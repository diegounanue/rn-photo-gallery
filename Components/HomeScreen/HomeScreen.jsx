import React, { useContext } from "react";
import {
  View,
  Text,
  TouchableOpacity,
  FlatList,
  Image,
  StyleSheet,
} from "react-native";
import { PhotoGalleryContext } from "../../Contexts/PhotoGalleryContext";

function HomeScreen({ navigation }) {
  const { photoGallery } = useContext(PhotoGalleryContext);

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.headerText}>List of Photos</Text>
      </View>
      <FlatList
        data={photoGallery}
        keyExtractor={(item, index) => index.toString()}
        numColumns={4}
        renderItem={({ item }) => (
          <View style={styles.photoItem}>
            <Image source={{ uri: item }} style={styles.photo} />
          </View>
        )}
      />
      <View style={styles.centeredButtonContainer}>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("Camera")}
        >
          <Text style={styles.buttonText}>Open Camera</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "white", // Color de fondo de la pantalla
  },
  header: {
    backgroundColor: "#7e75f8", // Color de fondo celeste
    padding: 10,
    alignItems: "center",
  },
  headerText: {
    color: '#fff',
    fontSize: 20
  },
  photoItem: {
    flex: 1,
    margin: 5,
  },
  photo: {
    width: 100,
    height: 100,
    resizeMode: "cover",
  },
  centeredButtonContainer: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  button: {
    borderWidth: 1,
    backgroundColor: "#dd57d3",
    width: 200,
    height: 50,
    justifyContent: "center", // Centra verticalmente el contenido del bot√≥n
    alignItems: "center",
  },
  buttonText: {
    color: '#fff'
  }
});

export default HomeScreen;
