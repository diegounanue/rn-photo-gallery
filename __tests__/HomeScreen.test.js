import React from 'react';
import HomeScreen from "../Components/HomeScreen/HomeScreen";
import renderer from 'react-test-renderer';

describe('HomeScreen Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HomeScreen navigation={{ navigate: jest.fn() }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
