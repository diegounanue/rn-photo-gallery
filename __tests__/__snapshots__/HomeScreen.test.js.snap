// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Component renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#7e75f8",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 20,
        }
      }
    >
      List of Photos
    </Text>
  </View>
  <RCTScrollView
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#dd57d3",
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": 200,
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
          }
        }
      >
        Open Camera
      </Text>
    </View>
  </View>
</View>
`;
